From fcde402ed711dc24883e207f5665fbb22eba6e4d Mon Sep 17 00:00:00 2001
From: anteater <65555601+nt8r@users.noreply.github.com>
Date: Mon, 14 Aug 2023 18:05:06 +0000
Subject: [PATCH] Do not abort when interfaces are older than expected

Instead, let consumers abort if the missing interface is a fatal problem.
---
 src/types/registry.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/src/types/registry.c b/src/types/registry.c
index b33c4d1..e1ea2c0 100644
--- a/src/types/registry.c
+++ b/src/types/registry.c
@@ -29,12 +29,14 @@
 
 #define BIND(interface_name, known_version) \
 if (strcmp(interface, #interface_name) == 0) { \
-    self->interface_name = wl_registry_bind( \
-        wl_registry, \
-        name, \
-        &interface_name ## _interface, \
-        known_version \
-    ); \
+    if (version >= known_version) { \
+        self->interface_name = wl_registry_bind( \
+            wl_registry, \
+            name, \
+            &interface_name ## _interface, \
+            known_version \
+        ); \
+    } \
 }
 
 static void wl_registry_global_handler(

